KhanhNT/linux-2020-7-2.md

```diff
@@ -0,0 +1,280 @@
+
+•	Redirecting:
+
+VD:   echo $LOGNAME >test.txt
+
+       nano test.txt hiển thị ra khanhnt
+
+Khi muốn chương trình nhận input từ file cho đến dòng tồn tại ký tự EOF (End of File), ta gõ lệnh:
+
+            file_name<<EOF
+
+•	Piping Date between programs
+Pipe data được sử dụng trong trường hợp ta muốn lấy output của program1 đem làm input của program2. Cách sử dụng như sau:
+
+$program 1 | program 2 | program 3
+
+
+## Processing text
+### cat, join, paste: 
+#### cat (concatenate): 
+
+cat <option> <ten file>
+
+cat <path file>
+
+cat <path file> <path file>:hiển thị nhiều nội dung file
+
+cat >ten file.txt: tạo tập tin rỗng hay xóa tập tin có sẵn
+
+cat -n <path file>: hiển thị số thứ tự của dòng
+
+cat file1.txt > file2.txt
+
+cat file1.txt >> file2.txt: lấy nd file 1 xuống cuối file 2
+
+cat file | less
+
+cat file | more
+
+#### join
+
+join file1 file2: gộp 2 file dưới dạng cột
+
+#### paste:
+-d: dấu phân cách thay vì tab
+
+-s: nối dữ liệu theo chuỗi, theo chiều ngang
+
+#### sort
+
+Sắp xếp các thứ tự theo thứ tự tăng dần hay giảm dần, theo bảng mã ASCII.
+
+sort <file>
+
+sort -r <file>: sắp xếp theo thứ tự ngược lại
+
+cat file1 file2 | sort
+==================================
+
+BEGIN {Actions}
+{Action} # Action for every line in a file
+END {Actions}
+
+#### uniq:
+Bỏ các dòng liên tiếp trong 1 văn bản trùng lặp.
+
+uniq -c filename: đếm số lượng thư mục trùng lặp
+
+
+#### head: 
+
+Dùng để xem dòng đầu của tập tin (mặc định là 10 dòng đầu)
+ 
+
+head <option> filename
+
+-n : chọn số dòng đầu hiển thị (n là số dòng)
+
+#### tail
+
+Dùng để xem dòng đầu tệp tin
+
+tail <option> filename
+
+-n: số dòng muốn xem
+
+-q: không in tiêu đề ra
+
+-f: tiếp tục đọc cho đến khi Ctrl+C
+
+#### wc:
+ 
+Dùng để đếm từ, byte, hoặc dòng
+
+wc <option> file 
+
+-l: hiển thị số dòng
+
+-c: hiển thị số lượng byte
+
+-w: hiển thị số lượng từ
+
+#### cut:
+
+cut kí tự của tệp
+
+cut -c a-b filename: lấy kí tự a đến b của file
+
+#### grep:
+
+grep <option> pattern <file>
+
+-i: tìm kiếm không quan trọng chữ hoa hay chữ thường
+
+-c: hiện số dòng khớp với mẫu tìm kiếm
+
+-r: tìm theo kiểu đệ quy
+
+-n: tìm dòng và tìm thông tin khớp với số dòng
+
+-v: hiển thị dòng không khớp với mẫu cần tìm
+
+grep query1 file | grep query2 file: tìm nhiều từ khóa khác nhau
+
+grep -l wordsearch ./*
+
+grep -e "chuoi" -e "chuoi" filename
+
+#### sed: stream editor
+
+sed <option> command <file>
+
+Lệnh in: sử dụng -n và p // sed -n 'từ cần lọc/p' 
+filename
+
+In và thay thế: sed -n 's/từ bị thay thế/từ thay thế/p' filename
+
+#### awk:
+
+awk '/search pattern 1/ {actions}/search pattern 2/{actions}'file
+
+awk '{print}' filename: in ra các dòng thuộc file
+
+awk '/098/{print}' file1.txt: in ra các dòng có chứa xâu mẫu 098 trong file1.txt
+
+## EDITOR FILE
+
+vi: trình soạn thảo văn bản ở trong terminal
+
+Cách mở: sudo vi <đường dẫn đến file cần chỉnh sửa>
+
+Chế độ command: di chuyển con trỏ bằng phím mũi tên. 
```

Comment:
có thể sử dụng các phím  hjkl để di chuyển

Reply:
https://superuser.com/questions/599150/why-arrow-keys-are-not-recommended-in-vim